care/emr/api/viewsets/encounter.py

```diff
@@ -252,6 +254,31 @@ def generate_discharge_summary(self, request, *args, **kwargs):
             status=status.HTTP_202_ACCEPTED,
         )
 
+    def get_permissions_for_encounter(self, user, encounter):
+        encounter_access = EncounterAccess()
+        roles = encounter_access.find_roles_on_encounter(user, encounter)
+        permissions = RolePermission.objects.filter(
+            role_id__in=roles, permission__context__in=["ENCOUNTER", "PATIENT"]
+        ).values_list("permission__slug", flat=True)
+        return list(permissions)
+
+    def retrieve(self, request, *args, **kwargs):
+        response = super().retrieve(request, *args, **kwargs)
+        instance = self.get_object()
+        response.data["permissions"] = self.get_permissions_for_encounter(
+            request.user, instance
+        )
+        return response
+
+    def list(self, request, *args, **kwargs):
```

Comment:
We dont need permissions for list, since there is only one action allowed, this API will also become extremely expensive , lets not do permissions for any list API